koku/koku/middleware.py

```diff
@@ -107,7 +106,11 @@ class KokuTenantSchemaExistsMiddleware(MiddlewareMixin):
     """A middleware to check if schema exists for Tenant."""
 
     def process_exception(self, request, exception):
-        if isinstance(exception, (Tenant.DoesNotExist, ProgrammingError)):
+        # double check if the Tenant exists
+        schema_name = request.user.customer.schema_name
+        try:
+            Tenant.objects.get(schema_name=schema_name)
+        except Tenant.DoesNotExist:
```

Comment:
Do we need to do any special handling of ProgrammingError? It looks like we [removed some important error handling code](https://github.com/project-koku/koku/commit/e06699cee3076a3cf55d6d9b79f34aa0e48b7fc8#diff-36209c7bd69cd5bf23b1960dc4297f06047d910797c7b954cc0dcc14421b706aL116) when settings were deprecated. Do we need to pull some of that back in minus the setting specific parts?

Reply:
What functionality would we need to pull back in?

Reply:
I don't know, really. I was just looking at the code that was removed a while back and wondering if that used to prevent us from swallowing the `ProgrammingError`. This is what was removed.

```python
           if (
                settings.ROOT_URLCONF == "koku.urls"
                and request.path in reverse("settings")
                and (not schema_exists(request.tenant.schema_name) or request.tenant.schema_name == "public")
            ):

                doc_link = generate_doc_link("/")

                err_page = {
                    "name": "middleware.settings.err",
                    "component": "error-state",
                    "errorTitle": "Configuration Error",
                    "errorDescription": f"Before adding settings you must create a Source for Cost Management. "
                    f"<br /><span><a href={doc_link}>[Learn more]</a></span>",
                }

                return JsonResponse(
                    [{"fields": [err_page], "formProps": {"showFormControls": False}}], safe=False, status=200
                )
```

Is any of that still useful or is what you have currently better overall? Trying to reason through that compound condinitonal statement is not fun.

Reply:
No, that bit was required when we were using the platform settings page. Since that was moved, this bit of code was no longer needed.

The issue of swallowing the ProgrammingErrors still existed even with this code in place.